### PR TITLE
[CR] Cutting up filthy clothes will yield filthy materials

### PR DESCRIPTION
**Pull request by Night-Pryanik**
The next step is obviously crafting with filthy materials.
Right now filthy materials are purely cosmetic as crafting from them will result in clean clothes.

I need opinions on what to do with filthy crafting.
- Forbid crafting from filthy materials entirely;
- Make crafted clothes filthy if even just one material needed to craft them is filthy;
- Make crafted clothes filthy if some percentage of materials is filthy (25%, 50%, 66% etc).

### CODE REVIEW

**Comment by Coolthulhu** [comment]
> I need opinions on what to do with filthy crafting.

Same problem as with rotten food.
Unfortunately, the best solution right now is to wait for a better crafting system that maps exact items by IDs rather than type IDs, then presents the choice to player.
Until then, you should probably disallow crafting with them and prevent filthy rag use to stop bleeding.

**Comment by Night-Pryanik** [comment]
> you should probably disallow crafting with them

This seems to have rather significant effect on gameplay. This means that players will have to stop cutting filthy clothes to get materials, especially high-tier like neoprene, kevlar, nomex and so on. And they must find another source of these materials.

**Comment by Coolthulhu** [comment]
Then keep the whole system as-is for now, except for dirty rags not being usable for bleeding.

Crafting filthy items must warn the player that they will be filthy too, do so before crafting is started, and not produce false positives (wrong: "this may be filthy if you craft it here", right: "this will be filthy if you use those components").

If we get item IDs, the crafting system will be easy to extend to allow asking the player. If...

**Comment by Night-Pryanik** [comment]
Ok, I don't change crafting; filthy rags ain't usable for healing now.
OTOH, all these "leather patch (filthy)" and "felt patch (filthy)" will be useless and just decorative.

**Comment by tinukedaya** [comment]
Hmm... Wouldn't it be better if filthy rags used to stop bleeding would cause infection instead of being forbidden to use? Not sure how hard to implement, but I can see me using filthy rag if it would save my life even if I know i'll get infected... After all, infection can be cured, death can't.

**Comment by kevingranade** [comment]
> Hmm... Wouldn't it be better if filthy rags used to stop bleeding would cause infection instead of being forbidden to use?

Following this reasoning, it would be an extreme and immediate infection, much worse than the infection you get from a bite.  The infection from a bite is due to a small amount of infectious material making its way into the wound, binding with a filthy bandage would not only press rotting material into the wound, but it would also keep it pressed into the wound over time, allowing the infection to take hold much more rapidly than normal.

IMO making filthy rags ineligible for use with healing is the most reasonable option, and it's certainly the most straightforward.

Speaking of options though, preventing filthy rags from being used to bind wounds should only happen if the filthy clothing mod is active.  i.e. add ```get_option<bool>( "FILTHY_WOUNDS" )``` to your check.

**Comment by kevingranade** [comment]
Short-term we could forbid crafting with filthy rags entirely, this would at least remain consistent, and you can wash them in order to use them.

**Comment by tinukedaya** [comment]
> Following this reasoning, it would be an extreme and immediate infection, much worse than the infection you get from a bite. The infection from a bite is due to a small amount of infectious material making its way into the wound, binding with a filthy bandage would not only press rotting material into the wound, but it would also keep it pressed into the wound over time, allowing the infection to take hold much more rapidly than normal.

That is of course correct and it should do exactly that. What I'm trying to say is that if I would need to choose between dying on spot bleeding out OR probably dying later on infection, I'll choose the later because I'd still have a chance to cure that infection.

**Comment by kolsurma** [comment]
There's a thought, if you've got a bottle of disinfectant could you not use it on a filthy rag to de-gunk it? 

Or perhaps first-aid should be a multi-step action, choose a sewing kit to close a wound, a rag to bind it, and pour disinfectant over the entire thing? Similar to UnReal World's system for major wounds.

**Comment by Coolthulhu** [comment]
> There's a thought, if you've got a bottle of disinfectant could you not use it on a filthy rag to de-gunk it?

This is pretty much the recipe for bandage.
Except the recipe for bandage also includes duct tape, while the rag is not held by anything (just held against the wound until it stops bleeding?).

**Comment by kolsurma** [comment]
>This is pretty much the recipe for bandage.

Well, yes. But, the bandage does more than just stop bleeding, it also heals a (very) tiny amount. And with the potential change to prevent filthy rags from being a crafting material for most things I think this would be a somewhat expensive way to get cleaned rags.

Another thought, what if it were a sliding scale? On one side you'd have the current Filthy (or gory) then dirty, clean, disinfected, and sterile? Instead of different items, you could use a float -1 to 1 or some such and pass it on to whatever it is used in, possibly with the lowest deciding overall cleanliness.

**Comment by kevingranade** [comment]
On Feb 19, 2017 5:38 PM, "Surma" <notifications@github.com> wrote:

Or perhaps first-aid should be a multi-step action, choose a sewing kit to
close a wound, a rag to bind it, and pour disinfectant over the entire
thing? Similar to UnReal World's system for major wounds.

This kind of thing is precisely what I want to happen eventually, but let's
not discuss it here, see
https://github.com/CleverRaven/Cataclysm-DDA/issues/16017 for the most
recent discussion.

**Comment by Night-Pryanik** [comment]
What's wrong with my latest commit?

**Comment by Coolthulhu** [comment]
Sounds like missing header for options.

**Comment by Night-Pryanik** [comment]
To forbid filthy crafting I need to somehow distinguish filthy rags from ordinary rags, right? Isn't that the same problem as in permitting filthy crafting that we touched in the beginning of this discussion?

**Comment by kevingranade** [comment]
I was opining about which approach would be better for the game, the issues
Coolthulu mention might still make it difficult to implement.  With the
option in place I think this pr is good to go, I just haven't had a chance
to test and merge.

File: src/iuse_actor.cpp [hunk 1]
```diff
@@ -2705,6 +2710,13 @@ long heal_actor::use( player *p, item *it, bool, const tripoint &pos ) const
         p->add_msg_if_player( m_info, _("You can't do that while underwater.") );
         return 0;
     }
+    
+    if( get_option<bool>( "FILTHY_WOUNDS" ) ) {
+        if( it->is_filthy() ) {
```
Comment (Coolthulhu) [inline src/iuse_actor.cpp]: Would be cleaner with `&&` instead of double-`if`.

**Comment by Coolthulhu** [comment]
> Isn't that the same problem as in permitting filthy crafting that we touched in the beginning of this discussion?

Now that filthy clothing has an option of opt-out, experimenting is fine. Most people who want filthy clothing in the current state will probably be fine with making it also limit crafting.

Does washing allow washing non-clothing items? If not, it should, so that materials from filthy items can be reclaimed.

**Comment by Night-Pryanik** [comment]
> Does washing allow washing non-clothing items?

Yes, it does. Clothing-only check is going through only when they drop from zombies.

**Comment by DangerNoodle** [comment]
One obvious question. Filthy clothing now seems to show up as filthy without the mod enabled, only having any actually effect if the Squeamish trait is taken.

In the case this behavior remains, it would absolutely be a good idea to ensure that the "prevent crafting with filthy items" feature does not apply without the filthy clothing mod, or this will again go back into forcing the effects of filthy clothing onto players whether they desire it or not.

**Comment by kevingranade** [comment]
Yes, basically all effects of filthy clothing should be supressed by the same mod.
If someone wants to follow up with a PR that replaces this with instant and severe infection, that would work too, keep in mind it needs to warn the player that this is going to happen

**Pull request merged**